static/skin/index.js

```diff
@@ -0,0 +1,229 @@
+        incrementalLoadingParams.count = viewPortToCount();
+        fadeOutDiv.style.display = 'none';
+        bookMap.clear();
+        params = new URLSearchParams(window.location.search);
+        if (filterType) {
+            params.set(filterType, filterValue);
+            window.history.pushState({}, null, `${window.location.href.split('?')[0]}?${params.toString()}`);
+        }
+        await loadAndDisplayBooks(true);
+    }
+
+    window.addEventListener('popstate', async () => {
+        await resetAndFilter();
+        filterTypes.forEach(key => {document.getElementsByName(key)[0].value = params.get(key) || ''});
+    });
+
+    async function loadSubset() {
+        if (window.innerHeight + window.scrollY >= document.body.offsetHeight) {
+            if (incrementalLoadingParams.count) {
+                loadAndDisplayBooks();
+            }
+            else {
+                fadeOutDiv.style.display = 'none';
+            }
+        }
+    }
+
+    window.addEventListener('resize', (event) => {
+        if (timer) {clearTimeout(timer)}
+        timer = setTimeout(() => {
+            incrementalLoadingParams.count = incrementalLoadingParams.count && viewPortToCount();
+            loadSubset();
+        }, 100, event);
+    });
+
+    window.addEventListener('scroll', loadSubset);
+
+    window.onload = async () => {
+        iso = new Isotope( '.book__list', {
+            itemSelector: '.book',
+            getSortData:{
+                weight: function( itemElem ) {
+                    const index = itemElem.getAttribute('data-idx');
+                    return index ? parseInt(index) : Infinity;
+                }
+            },
+            sortBy: 'weight'
+        });
+        footer = document.getElementById('kiwixfooter');
+        fadeOutDiv = document.getElementById('fadeOut');
+        await loadAndDisplayBooks();
+        await loadAndDisplayOptions('#languageFilter', langList);
+        await loadAndDisplayOptions('#categoryFilter', categoryList);
+        filterTypes.forEach((filter) => {
+            const filterTag = document.getElementsByName(filter)[0];
+            filterTag.addEventListener('change', () => {resetAndFilter(filterTag.name, filterTag.value)});
+        });
+        params.forEach((value, key) => {document.getElementsByName(key)[0].value = value});
+        document.getElementById('kiwixSearchForm').onsubmit = (event) => {event.preventDefault()};
+        if (!window.location.search) {
+            const browserLang = navigator.language.split('-')[0];
+            if (browserLang.length === 3) {
+                document.getElementById('languageFilter').value = browserLang;
+                langFilter.dispatchEvent(new Event('change'));
+            } else {
+                const langFilter = document.getElementById('languageFilter');
+                langFilter.value = iso6391To3[browserLang];
+                langFilter.dispatchEvent(new Event('change'));
+            }
```

Comment:
This can be simplified to:

```js
const langFilter = document.getElementById('languageFilter');
langFilter.value = browserLang.length === 3 ? browserLang : iso6391To3[browserLang];
langFilter.dispatchEvent(new Event('change'));
```